scripts/linters/general_purpose_linter.py

```diff
@@ -694,203 +683,142 @@ def _check_for_mandatory_pattern_in_file(
+        error_messages = []
         files_to_lint = self.all_filepaths
         failed = False
 
-        with linter_utils.redirect_stdout(sys.stdout):
-            for filepath in files_to_lint:
-                file_content = self.file_cache.readlines(filepath)
-                file_length = len(file_content)
-                if (
-                        file_length >= 1 and
-                        not re.search(r'[^\n]\n', file_content[-1])):
-                    summary_message = (
-                        '%s --> There should be a single newline at the '
-                        'end of file.' % filepath)
-                    summary_messages.append(summary_message)
-                    python_utils.PRINT(summary_message)
-                    failed = True
-
-            if failed:
-                summary_message = (
-                    '%s Newline at the eof check failed.' % (
-                        linter_utils.FAILED_MESSAGE_PREFIX))
-            else:
-                summary_message = (
-                    '%s Newline at the eof check passed.' % (
-                        linter_utils.SUCCESS_MESSAGE_PREFIX))
-            summary_messages.append(summary_message)
-            python_utils.PRINT(summary_message)
-
-        return summary_messages
+        for filepath in files_to_lint:
+            file_content = self.file_cache.readlines(filepath)
+            file_length = len(file_content)
+            if (
+                    file_length >= 1 and
+                    not re.search(r'[^\n]\n', file_content[-1])):
+                error_message = (
+                    '%s --> There should be a single newline at the '
+                    'end of file.' % filepath)
+                error_messages.append(error_message)
+                failed = True
+        full_error_messages = error_messages
+        return concurrent_task_utils.TaskResult(
+            name, failed, error_messages, full_error_messages)
 
     def perform_all_lint_checks(self):
         """Perform all the lint checks and returns the messages returned by all
         the checks.
 
         Returns:
-            str. All the messages returned by the lint checks.
+            list(TaskResult). A list of TaskResult objects to be used for
+            linter status retrieval.
         """
-        mandatory_patterns_messages = self._check_mandatory_patterns()
-        pattern_messages = self._check_bad_patterns()
-        newline_at_eof_messages = self._check_newline_at_eof()
-
-        all_messages = (
-            mandatory_patterns_messages + pattern_messages +
-            newline_at_eof_messages)
-        return all_messages
+        if not self.all_filepaths:
+            concurrent_task_utils.log('There are no files to be checked.')
+        linter_stdout = []
+        linter_stdout.append(self.check_mandatory_patterns())
+        linter_stdout.append(self.check_bad_patterns())
+        linter_stdout.append(self.check_newline_at_eof())
```

Comment:
```suggestion
        linter_stdout = [
        self.check_mandatory_patterns(),
        self.check_bad_patterns(),
        self.check_newline_at_eof()]
```

Should we do it this way^? (Fine with either!)

Reply:
Done.